### ISSUE TITLE
Log when host/worker is ready for VS Code debugger to attach

### ISSUE THREAD

**Issue by anthonychu** [Azure/azure-functions-host#6252]
After discussing with @pragnagopa, it sounds like the initial change in #6230 is not quite sufficient for the purposes of VS Code attaching the debugger reliably.

There are 2 issues in particular:
- In-proc .NET apps will not print the new message
- The message is printed with log level Information. If default log levels are changed to Warning, Error, or None, the message will not print

We need this message to be printed regardless of stack or log level. Also suggest updating the message to something more generic and doesn't mention "worker process". (something like "Function app is ready"?)

Also needed for this: https://github.com/Azure/azure-functions-core-tools/issues/1131

@ericjizbaMSFT

**Comment by pragnagopa** [comment]
> If default log levels are changed to Warning, Error, or None, the message will not print

New log is added at info level. Setting to any other level will include info level logs. This would break if the user turns off logging completely.

**Comment by anthonychu** [comment]
@ericjizbaMSFT Is there another way to attach a debugger in VS Code without relying on output?

@pragnagopa Can the host `Console.WriteLine` the message (bypassing any configured log levels) if it's running inside Core Tools?

**Comment by pragnagopa** [comment]
@brettsam  - As of now VS Code relies on logging to attach a debugger. If the user sets logLevel to `None`, is there way to still write logs to console when running in core tools?

**Comment by EricJizbaMSFT** [comment]
> is there a way VS Code can attach debugger without relying on the logging?

Well we only use the logs for languages that use a port for debugging, like Python/JS/etc. because we can't attach until the port is open. For C#, we attach directly to the process and we don't need to wait

> This would break if the user turns off logging completely.

Do you by chance log the log level? Aka do you print a line that says "Log level: 'None'" or "Logs will not be printed because log level is set to 'None'"? If so, we can detect _that_ and throw an error telling users we need logs

**Comment by pragnagopa** [comment]
@EricJizbaMSFT - thanks for clarifying about C# vs out-of-proc languages. Fix in #6230 is enough to address the issue on when attach debugger.

@anthonychu  - Min loglevel is set to information. Setting log level to None in host.json

```
"Logging": {
		"LogLevel": {
			"Default": "None"
		}
	}
```
does not impact info level logs.

Do you have an example how attaching debugger fails due to log level?

**Comment by anthonychu** [comment]
@pragnagopa Pushed a repro [here](https://github.com/anthonychu/20200630-test-loglevels). I basically made the change in host.json you mentioned. It suppresses the current "Host lock lease acquired" message. I don't have the a build with the updated host with the new message. Perhaps that's not affected by the log level?

**Comment by EricJizbaMSFT** [comment]
> Well we only use the logs for languages that use a port for debugging, like Python/JS/etc. because we can't attach until the port is open. For C#, we attach directly to the process and we don't need to wait

So I did more digging and turns out this isn't the case _yet_. I think it's possible, but still need to do some more investigating and work to flesh it out. If it's easy for you to add the log in the .NET case, that would be great - we would "just work" with little effort from my side. If that's not easy, I'm comfortable saying the .NET case is on our side to fix.

On another note - we have a release coming up in a week-ish and I plan to update our pattern in that release. @pragnagopa sound good to you? I plan to maintain backwards compatibility and keep looking for the old message, too. Once people update to the new cli, it should see the new message first and attach a little bit sooner (ideally fixing the runOnStartup issues)

**Comment by pragnagopa** [comment]
@EricJizbaMSFT  - For C#, can you continue to rely on existing pattern `Host lock lease acquired...` ? For all other languages you can use new log message.  

> On another note - we have a release coming up in a week-ish and I plan to update our pattern in that release.
Release that includes the new log messages is not rolled out yet. But since your changes would be backward compatible, you do not have to block on host release.

@anthonychu - I can repro the issue. Setting log level to `None` does hide all info logs coming from JobHost. Tested with latest bits. 
Following code applies the filter:

https://github.com/Azure/azure-functions-host/blob/c17b336f845dd24e9b8e80613535bd5333c47b3a/src/WebJobs.Script/Extensions/ScriptLoggingBuilderExtensions.cs#L36

I will dig a little more to figure out if we can enforce log out put for logs VS Code relies on even when logLevel is set to None.

Updating issue title and assigning to next sprint.

**Comment by EricJizbaMSFT** [comment]
> For C#, can you continue to rely on existing pattern Host lock lease acquired... ? For all other languages you can use new log message.

Yes we can. It doesn't fix the "runOnStartup" [issue for C#](https://github.com/microsoft/vscode-azurefunctions/issues/1971), but it maintains the current experience.
